src/fix_gemc.cpp

```diff
@@ -0,0 +1,514 @@
+// clang-format off
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   https://www.lammps.org/, Sandia National Laboratories
+   LAMMPS development team: developers@lammps.org
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+// Gen Questions:
+// 1) What is a tagint?
+// 2) What are all the nullptrs in the definition of a fix
+
+/* ----------------------------------------------------------------------
+   Contributing author: Paul Crozier, Aidan Thompson (SNL)
+------------------------------------------------------------------------- */
+
+#include "fix_gemc.h"
+
+#include "atom.h"
+#include "atom_vec.h"
+#include "comm.h"
+#include "compute.h"
+#include "domain.h"
+#include "error.h"
+#include "force.h"
+#include "group.h"
+#include "input.h"
+#include "memory.h"
+#include "modify.h"
+#include "molecule.h"
+#include "neighbor.h"
+#include "pair.h"
+#include "random_park.h"
+#include "universe.h"
+#include "update.h"
+#include "variable.h"
+
+// for molecule
+#include "angle.h"
+#include "bond.h"
+#include "dihedral.h"
+#include "improper.h"
+#include "kspace.h"
+
+#include <cstring>
+
+using namespace LAMMPS_NS;
+using namespace FixConst;
+
+// Notes:
+// 1) full flag always on
+
+// large energy value used to signal overlap
+
+static constexpr double MAXENERGYSIGNAL = 1.0e100;
+
+// this must be lower than MAXENERGYSIGNAL
+// by a large amount, so that it is still
+// less than total energy when negative
+// energy contributions are added to MAXENERGYSIGNAL
+
+static constexpr double MAXENERGYTEST = 1.0e50;
+
+static constexpr double BUFFACTOR = 1.2;
+
+// const std::vector<std::string> AtomVec::default_exchange = {"id",    "type", "mask",
+//                                                            "image", "x",    "v"};
+// bufextra
+static constexpr int BUFMIN = 1024;
+
+/* ---------------------------------------------------------------------- */
+
+// not sure what all these nullptrs are for
+FixGEMC::FixGEMC(LAMMPS *lmp, int narg, char **arg) : Fix(lmp, narg, arg)
+{
+  buf = NULL;
+
+  if (narg < 12) utils::missing_cmd_args(FLERR, "fix gemc", error);
+  // must have only two boxes
+
+  if (universe->nworlds != 2) error->universe_all(FLERR, "Must use exactly two partitions");
```

Comment:
Since this fix implements both, Monte Carlo and a multi-partition fix, it would need to be added to either the MC package or the REPLICA package. My preference is for the REPLICA package. For sure the source code cannot remain in the src folder. This is reserved for core functionality.

Reply:
Fixed.